firstvoices/backend/serializers/widget_serializers.py

```diff
@@ -36,3 +36,46 @@ class Meta(SiteContentLinkedTitleSerializer.Meta):
             "format",
             "settings",
         )
+
+
+class SiteWidgetListOrderDetailSerializer(SiteWidgetDetailSerializer):
+    order = serializers.SerializerMethodField()
+
+    class Meta(SiteWidgetDetailSerializer.Meta):
+        model = SiteWidget
+        fields = (
+            "id",
+            "order",
+            "title",
+            "widget_type",
+            "format",
+            "visibility",
+            "settings",
+        )
+
+    def get_order(self, widget):
+        if "homepage" in self.context:
+            return (
+                widget.sitewidgetlistorder_set.filter(
+                    site_widget_list=self.context["homepage"]
+                )
+                .first()
+                .order
+            )
+        return None
+
+
+class SiteWidgetListSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = SiteWidgetList
+
+    def to_representation(self, instance):
+        widgets = instance.widgets.all()
+        unordered_widgets = SiteWidgetListOrderDetailSerializer(
+            widgets, many=True, context=self.context
+        ).data
+        ordered_widgets = sorted(unordered_widgets, key=lambda x: x["order"])
+        ordered_widgets = list(
+            filter(lambda x: x.pop("order", None) or True, ordered_widgets)
+        )
```

Comment:
I think you could accomplish this by adding an order_by clause to the prefetch in the view instead. Might be simpler.